server/src/com/cloud/storage/StorageManagerImpl.java

```diff
@@ -1830,12 +1830,37 @@ public boolean storagePoolHasEnoughSpace(List<Volume> volumes, StoragePool pool,
             }
         }
 
+        return checkPoolforSpace(pool, allocatedSizeWithTemplate, totalAskingSize);
+    }
+
+    @Override
+    public boolean storagePoolHasEnoughSpaceForResize(StoragePool pool, long currentSize, long newSiz) {
+        if (!checkUsagedSpace(pool)) {
+            return false;
+        }
+        if (s_logger.isDebugEnabled()) {
```

Comment:
Is this if really needed? Can't you just log to debug and let the filter of log4j handle it?

Reply:
If log4j does not have debug enabled, then the string concat operation will still be performed by the JVM only when the string is passed to the `debug()` method it will be ignored. So, it depends if we can afford the CPU/time spent on the operation.

Reply:
I think we can afford this tiny bit of extra CPU.

Reply:
I think as a matter of principle we should alwas add the check. You never no for what reason someone will alter the message. I like using String.format to construct the message, btw, and then pass the single resulting string to the logger

Reply:
What principle?

I also use String.format, and I do not see the problem.

Reply:
"bla" + "blob" is a costly operation, basically to memcpy()s. String.format() as well in comparison. We don't want those operations performed if we don't end up using the result in a log statement. so the boolean check potentially saves us a lot of cpu cycles, especially in the debug() and trace() cases.

Reply:
Yes, I do know that. However, we should ask ourselves, if the addition of a if conditions (it will clutter the code) is worth the CPU cycles. For me, I would not trade readability for this small CPU cost in a web system (ACS is a at the end of the day a Java web application).